test/membrane_rtc_engine/hls_endpoint_test.exs

```diff
@@ -0,0 +1,118 @@
+defmodule Membrane.RTC.HLSEndpointTest do
+  use ExUnit.Case
+
+  alias Membrane.RTC.Engine
+  alias Membrane.RTC.Engine.Endpoint.HLS
+  alias Membrane.RTC.Engine.{Message, Peer}
+  alias Membrane.RTC.Engine.Support.FileEndpoint
+
+  @fixtures_dir "./test/fixtures/"
+  @reference_dir "./test/hls_reference/"
+  @output_dir "./test/hls_output/"
+
+  setup do
+    options = [
+      id: "test_rtc"
+    ]
+
+    {:ok, pid} = Engine.start_link(options, [])
+
+    Engine.register(pid, self())
+
+    [rtc_engine: pid]
+  end
+
+  describe "HLS Endpoint test" do
+    test "FileEndpoint send media to HLS Endpoint", %{rtc_engine: rtc_engine} do
+      peer_id = "file-endpoint-peer"
+      metadata = %{"display_name" => "test_peer"}
+
+      add_peer(rtc_engine, peer_id, metadata)
```

Comment:
I would treat file endpoint as standalone, server endpoint i.e. endpoint that doesn't correspond to any peer

Reply:
As we need to send to endpoint `custom_media_event` we need to have an associate peer with this endpoint.